### ISSUE TITLE
Using any torchvision pretrained model as backbone for FasterRcnn

### ISSUE THREAD

**Issue by mrinath123** [pytorch/vision#6172]
### 🚀 The feature

Adding only the name of the backbone network, loads the pretrained model in the FasterRcnn model as backbone.

### Motivation, pitch

Its possible to use any backbone, but we have to specify many things like feature size, feature_name etc., wont be it easy for the user to just give the name of the backbone and we internally do everything.

**Comment by mrinath123** [comment]
Need reviews for this, and thanks for listening to my idea
CC @datumbox

**Comment by datumbox** [comment]
Thanks for the recommendation. If I understand you correctly, you would like to make seamless the combination of different backbones with different detection heads, is that right?

The current API, allows you to provide a `backbone` on the constructor of the Detection models. This is the first argument that all Object Detection classes receive. Using it is a rather advanced API because one has to pass quite a few configurations to it such as AnchorGenerators. How one will configure these settings is non-trivial and often backbone and dataset specific. Some models even need to adjust the anchors depending on the input size. Here is an investigation we did previously around this which shows how sensitive this is: https://github.com/pytorch/vision/issues/3819#issuecomment-848040195. As a result it's rather difficult to provide one generic method that can attach any backbone to any head and expect it to work (at least that's true for Anchor-based techniques with FPNs etc).

This is the reason we only offer a limited number of models for Detection paired with specific backbones. Typically those are the backbones that their original paper used but sometimes we bended this rule (for example `fasterrcnn_mobilenet_v3_large_fpn`) to provide better support to for mobile specific applications. Having said that, there is interesting work that researchers are doing in the field to simplify the detection task. For example see [this paper](https://arxiv.org/abs/2203.16527) by @lyttonhao et al.

So your proposal makes sense, but it's quite hard to achieve with the current models. But it's something that there is active research on and might become possible sooner than later. If I misunderstood the question, please ignore the above and let me know what I missed. Thanks! :)

**Comment by mrinath123** [comment]
Yes, exactly I was talking about the same thing. Like you said its not easy to like know some internal config. I think if we can solve this issue, it will be really helpful for the users.
If any work related to this comes I would definitely want to contribute.
Thanks for making it know why its a complex process and giving the link of the papers, I will try to study them.

**Comment by mrinath123** [comment]
Nice repo!
I too made something similar https://github.com/mrinath123/timmFasterRcnn I used timm backbone models rather than pytorch vision, but as you can see in the model config, there is a lot of manual work to do.
I hope there will be a easier solution to do this.
